### PR TITLE
additional themed components

### PR DESCRIPTION
**Pull request by hpwBLN**
_Attribution: Donated by The Library Code_

## Description
This PR makes more components themeable and fixes some code.

## Instructions for Reviewers

1. This PR makes the following components themeable:

- Search-navbar component
- Comcol-page-handle component
- Object-list component
- Collection-list-element component
- Community-list-element component
- Browse-by-date-page component
- Browse-by-metadata-page component
- Browse-by-title-page component

2. It adds code of already themed Comcol-page-browse-by component to the custom theme.

3. It fixes a forgotten "implements OnDestroy" in browse-by-metadata-page.component.ts

5. It fixes some involved unit tests to wait for async functions to finish before evaluating outcomes (thank you, @kshepherd) .

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by tdonohue** [comment]
@hpwBLN : This has merge conflicts now because of recent code mergers to `main`.  Could you give it a quick rebase when you get the chance?  Then we can get this reviewed & ready to go.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @hpwBLN !  This looks good to me. I've also tested these newly themed components using both the `custom` and `dspace` themes and haven't found any issues.

I'll leave this open briefly to let @artlowel give it a quick look as well, but I think this is likely ready to be merged.

**Comment by hpwBLN** [comment]
Thank you @tdonohue, we are happy to coontribute.

**Review by artlowel — COMMENTED** [review]
Thanks @hpwBLN! 

It works well, I just have 2 minor inline comments

File: src/themes/custom/app/browse-by/browse-by-title-page/browse-by-title-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Component } from '@angular/core';
+import { BrowseByTitlePageComponent as BaseComponent } from '../../../../../app/browse-by/browse-by-title-page/browse-by-title-page.component';
+
+@Component({
+  selector: 'ds-browse-by-title-page',
+  // styleUrls: ['./browse-by-title-page.component.scss'],
+  styleUrls: ['../../../../../app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.scss'],
+  // templateUrl: './browse-by-title-page.component.html'
```
[the remark this answers is not in the document]
Comment (hpwBLN) [reply src/themes/custom/app/browse-by/browse-by-title-page/browse-by-title-page.component.ts:8]: Thank you @artlowel , I created the missing template files.

**Review by artlowel — APPROVED** [review]
Thanks!

**Pull request merged**